Krop Aliaksei

alekseikrop opened the pull request.

Commit:
Initial commit

Created command line interface with argparse module.

Commit:
RssHandler class methods realisation

- RssHandler class was added to a separate .py file
- Added realisation of url_handler and rss_dict methods

Commit:
base functionality

Added base functionality of RSS reader
- Created output method, which prints result in stdout
- Added realisation to to_json method

Commit:
Verbose realisation

Added realisation for verbose option with logging library.
Completed first iteration of final task

Commit:
Distribution

-Added setup.py script.
-Added LICENSE.txt, MANIFEST.in, README.md, requirements.txt files.

Commit:
Fix distribution

Fixed error with distribution

AlexeiBuzuma commented on FinalTask/RSSHandle.py:
```diff
@@ -0,0 +1,78 @@
+class RssHandler:
+
+    def __init__(self, rss: str):
+        self.rss = rss
```
Если посмотреть по коду, то сюда приходит нe RSS, а ссылка, по которой можно забрать RSS
это именование может сильно запутать, советую переименовать этот атрибут, чтобы было более очевидно его содержимое

AlexeiBuzuma commented on FinalTask/rss_reader.py:
```diff
@@ -0,0 +1,46 @@
+    rss_object.output(args.json, args.limit)
+
+
+
```
Должно быть 2 пустые строки, не 3

AlexeiBuzuma commented on FinalTask/setup.py:
```diff
@@ -0,0 +1,16 @@
+import setuptools
+
+setuptools.setup(
```
консольная утилита не создается для rss ридера
Фактически нет возможности проверить работоспособность решения без написания кода на пайтоне

```bash
root@1ebd15600cbe:/PythonHomework/FinalTask# rss_reader
bash: rss_reader: command not found
root@1ebd15600cbe:/PythonHomework/FinalTask# rss-reader
bash: rss-reader: command not found
root@1ebd15600cbe:/PythonHomework/FinalTask#
```

AlexeiBuzuma commented on FinalTask/RSSHandle.py:
```diff
@@ -0,0 +1,78 @@
+from bs4 import BeautifulSoup
```
pycodestyle выдает несколько стилистических ошибок
```bash
# pycodestyle --max-line-length=120 .
./rss_reader.py:7:1: E302 expected 2 blank lines, found 1
./rss_reader.py:45:1: E303 too many blank lines (3)
./setup.py:12:24: E251 unexpected spaces around keyword / parameter equals
./setup.py:12:26: E251 unexpected spaces around keyword / parameter equals
./setup.py:14:23: E231 missing whitespace after ':'
./test.py:4:1: E302 expected 2 blank lines, found 1
```

Commit:
Fix issues

- Fixed setup issues;
- Fixed code style issues;
- Changed packaging of project;
- Added new class and few methods in it for cache control.

Commit:
Iteration 2

- Updated requirments.txt file;
- Added dependencies to the setup.py file

Commit:
Base cache

Cache now stored in sqlite database

Commit:
Fix db

Fixed cache database realisation

Commit:
DB path

Added db path and changed packaging

Commit:
Finished iteration 3

News caching was implemented. Feed now stored in sqlite database and can be accesed with --date argument.

Commit:
Cache JSON

Added possibility to convert feed from cache to JSON format.

Commit:
Convert to html.

News now can be converted and stored in html format.

Commit:
Updated requirments

Updated requirments.txt file and requirments in setup.py

AlexeiBuzuma commented:
Не устанавливается 
```bash
# pip install .
Processing /data/PythonHomework/FinalTask
    ERROR: Command errored out with exit status 1:
     command: /usr/local/bin/python -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-req-build-1l51ht8m/setup.py'"'"'; __file__='"'"'/tmp/pip-req-build-1l51ht8m/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-req-build-1l51ht8m/pip-egg-info
         cwd: /tmp/pip-req-build-1l51ht8m/
    Complete output (1 lines):
    error in rss_reader setup command: 'install_requires' must be a string or list of strings containing valid project/version requirement specifiers; Invalid requirement, parse error at "'=2.10.3'"
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
```

AlexeiBuzuma commented on FinalTask/rss_reader_module/rss_reader.py:
```diff
@@ -0,0 +1,89 @@
+                       type=str,
+                       help='Print cached news that was published on given date')
+
+    parser.add_argument('--to-html',
```
Конвертация реализована только в один формат, верно?

alekseikrop replied:
Верно. В fb2 реализовано, но работает не правильно.

Commit:
Fixed setup error.

Fixed setup error

AlexeiBuzuma commented:
Пункты, которые могут быть улучшены:
1) Коммит сообщения не всегда передают происходящее, например "Fix issues" и "DB path"
2) При использовании базы данных есть моменты, где не будет закорыто соединение к базе (например файл RSSHandle строка 117 никогда не будет вызвана)
3) В коде есть магические числа (например файл RSSHandle строка 123)
4) Файл RSSHandle, стркоа 126. Есть опасность с возможной SQL инъекцией (https://ru.wikipedia.org/wiki/%D0%92%D0%BD%D0%B5%D0%B4%D1%80%D0%B5%D0%BD%D0%B8%D0%B5_SQL-%D0%BA%D0%BE%D0%B4%D0%B0)
5) Не везде есть док стринги. Местами они были бы очень уместны.
6) нет юнит тестов

Из хорошего могу отменить, что в основном код хорошо разбит на функции и в целом не возникает больших трудностей, чтобы понять, что делает та или иная функция.
